### DISCUSSION TITLE
How do I make a request to support new artifact kind.

### DISCUSSION THREAD

**Discussion by zong-zhe** [artifacthub/hub#3257] [General]
Hi ! How to make a new request for `artifacthub.io` to support new artifact kind, I can not find guide on it.  

We have created a new configuration language called KCL and we expect to use `artifacthub.io` as a registry for our KCL artifacts.

And: 
- Are there any conditions for  `artifacthub.io` to support `KCL artifact` ?
- How to make the request ?

For more information about the KCL: https://kcl-lang.io/

**Comment by tegioz** [comment]
Hi @zong-zhe 👋 

Artifact Hub currently only supports vendor neutral artifacts. Everything listed is currently in a foundation and not attached to a single vendor. This is the main requirement for new artifacts kinds.

I've seen that [you've applied to join the CNCF foundation as a sandbox project](https://github.com/cncf/sandbox/issues/48). So as soon as your project is accepted you'd be ready to request support for the new artifact kind. To do that, you can file an issue in this repository and we'll take care of it 🙂

**Reply by zong-zhe** [reply]
Thanks @tegioz, this is helpful, we will ask for the KCL artifact kind support after we are accepted. 😀

**Comment by zong-zhe** [comment]
Hello @tegioz , thank you very much for your earlier reply.  😀

Our project has entered the voting phase for joining the CNCF sandbox https://github.com/cncf/sandbox/issues/48. I think I can begin to prepare the issue. I have a few questions.

- Are there guidelines for what I need to prepare ?

- If we want to use `artifacthub.io` as a KCL artifacts market for the package management, are there any requirements for the content of the artifacts ?

- Is there a template for the issue ? Is this one `Official status request` ?

**Reply by tegioz** [reply]
Hi @zong-zhe

No worries! Not really, just expressing your interest and providing some details about your artifacts should be enough to get the process started 🙂

**Comment by zong-zhe** [comment]
Hi @tegioz 😀
We are grateful for AH’s support towards KCL, and we are very satisfied with AH. We have a slight adjustment that we would like to propose. After our discussion, we believe that changing the kind of KCL artifacts from ‘KCL Packages’ to ‘KCL Modules’ would be more appropriate.

And, a small question is 
- when can we see support for KCL on `artifacthub.io` ?

**Reply by tegioz** [reply]
We're glad you like it!

`KCL packages` have been renamed to `KCL modules` and they are now available on <https://artifacthub.io> 🙂
